bin/plugin/commands/release.js

```diff
@@ -69,40 +69,59 @@ async function runReleaseBranchCreationStep(
 	gitWorkingDirectoryPath,
 	abortMessage
 ) {
-	let version, releaseBranch, versionLabel;
+	let parsedVersion, releaseBranch, versionLabel;
 	await runStep( 'Creating the release branch', abortMessage, async () => {
 		const packageJsonPath = gitWorkingDirectoryPath + '/package.json';
 		const packageJson = readJSONFile( packageJsonPath );
-		const nextMajor = getNextMajorVersion( packageJson.version );
-		const parsedMajorVersion = semver.parse( nextMajor );
+		const packageVersion = semver.parse( packageJson.version );
+		const isPackageVersionRC = packageVersion.prerelease.length > 0;
 
-		version = nextMajor + '-rc.1';
+		// if the last tag is an RC then we can assume that we should generate a new rc with the same version number
+		if ( isPackageVersionRC ) {
+			parsedVersion = semver.parse(
+				semver.inc( packageVersion.version, 'prerelease', 'rc' )
+			);
+		} else {
+			const nextMajor = getNextMajorVersion( packageJson.version );
+			parsedVersion = semver.parse( nextMajor + '-rc.1' );
+		}
+
+		versionLabel = parsedVersion.raw.replace( /\-rc\.([0-9]+)/, ' RC$1' );
```

Comment:
Let's maybe use a (non-parsed) `version` as the variable, since we need that more often than the parsed one:

```suggestion
		if ( isPackageVersionRC ) {
			version = semver.inc( packageVersion.version, 'prerelease', 'rc' );
		} else {
			const nextMajor = getNextMajorVersion( packageJson.version );
			version = nextMajor + '-rc.1';
		}

		versionLabel = version.replace( /\-rc\.([0-9]+)/, ' RC$1' );
```